internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -185,9 +185,26 @@ func (h *Diagnostics) collectDiag(ctx context.Context, action *fleetapi.ActionDi
 }
 
 // runHooks runs the agent diagnostics hooks.
-func (h *Diagnostics) runHooks(ctx context.Context) ([]client.DiagnosticFileResult, error) {
+func (h *Diagnostics) runHooks(ctx context.Context, action *fleetapi.ActionDiagnostics) ([]client.DiagnosticFileResult, error) {
 	hooks := append(h.diagProvider.DiagnosticHooks(), diagnostics.GlobalHooks()...)
-	diags := make([]client.DiagnosticFileResult, 0, len(hooks))
+
+	// Currently CPU is the only additional metric we can collect.
+	// If this changes we would need to change how we scan AdditionalMetrics.
+	collectCPU := false
+	for _, metric := range action.AdditionalMetrics {
+		if metric == "CPU" {
```

Comment:
Can the agent import the generated Fleet API modules? If it can't, why not?

I think we are also duplicating the upgrade details definitions in the agent which feels wrong.

Reply:
Yes, it should be able to as fleet-server exports [a client and type definitions](https://github.com/elastic/fleet-server/tree/main/pkg/api). However we will need to refactor in order to use it, I've created an issue to track it [here](https://github.com/elastic/elastic-agent/issues/4400)